python/tvm/tir/op.py

```diff
@@ -3037,6 +3037,148 @@ def TVMBackendFreeWorkspace(device_type, device_id, ptr):
     return call_intrin("int32", "tir.TVMBackendFreeWorkspace", device_type, device_id, ptr)
 
 
+# Intrinsics for the VK_NV_cooperative_matrix Vulkan extension.
+
+
+def cooperative_matrix_load_NV(buffer_mat, offset, src, rows, cols, stride, column_major):
```

Comment:
The parameters for this intrinsic look very similar to those of [BufferRegion](https://github.com/apache/tvm/blob/main/include/tvm/tir/stmt.h#L1099), but I can't think of any way to use it in this context.  If we could, that would be a nice way to name as many parameters as possible.

Reply:
What do you mean? I don't see a similarity.

Reply:
Mostly that the parameters used for the intrinsic are specifying a source and destination region for a copy to be performed.  There are some restrictions to it, such as requiring that the source and destination buffers are both 2-d, and that one of the two values in `source_region->buffer->strides` is one.  I was thinking that it might be a cleaner representation to write the intrinsic as `cooperative_matrix_load_NV(dest_region, source_region)`, if possible.

Looking at it again this morning, though, writing this intrinsic in terms of BufferRegion would only work if the buffers are maintained in their non-flattened state all the way through to codegen, as determining the rows/cols/stride/column_major parameters would require a 2-d buffer.  That restriction, along with needing to compute some of the strides during codegen, is making me like the original implementation more, with the intrinsic directly matching the SPIR-V opcode.